### PR TITLE
[2024-08-08] Bump dependency updates identified by dependabot

### PR DESCRIPTION
**Pull request by henrybear327**
Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

This pull request completes this week's etcd dependency updates following our [dependency roster](https://docs.google.com/spreadsheets/d/1jodHIO7Dk2VWTs1IRnfMFaRktS9IH8XRyifOnPdSY8I/edit#gid=1394774387) and [dependency management instructions](https://github.com/etcd-io/etcd/blob/main/Documentation/contributor-guide/dependency_management.md). 

Summary of actions

- Bumped
    - bump [golang.org/x/sys](http://golang.org/x/sys) from 0.22.0 to 0.23.0
        - https://github.com/etcd-io/etcd/pull/18406
        - https://github.com/etcd-io/etcd/pull/18402
    - bump [golang.org/x/sync](http://golang.org/x/sync) from 0.7.0 to 0.8.0
        - https://github.com/etcd-io/etcd/pull/18405
    - bump [golang.org/x/time](http://golang.org/x/time) from 0.5.0 to 0.6.0
        - https://github.com/etcd-io/etcd/pull/18401
    - bump github.com/alexfalkowski/gocovmerge from v1.3.9 to v1.3.13 
        - https://github.com/etcd-io/etcd/pull/18418#discussion_r1709929410
    - bump golang.org/x/crypto from 0.25.0 to 0.26.0
        - Due to bump github.com/alexfalkowski/gocovmerge from v1.3.9 to v1.3.13 
    - bump golang.org/x/net from 0.27.0 to 0.28.0
        - Due to bump github.com/alexfalkowski/gocovmerge from v1.3.9 to v1.3.13 
    - bump golang.org/x/text from 0.16.0 to 0.17.0
        - Due to bump github.com/alexfalkowski/gocovmerge from v1.3.9 to v1.3.13 
- Closed (fully indirect)
    - bump [github.com/zmap/zlint/v3](http://github.com/zmap/zlint/v3) from 3.5.0 to 3.6.3
        - https://github.com/etcd-io/etcd/pull/18407
    - bump [github.com/mgechev/revive](http://github.com/mgechev/revive) from 1.3.7 to 1.3.9
        - https://github.com/etcd-io/etcd/pull/18404
    - bump [golang.org/x/mod](http://golang.org/x/mod) from 0.18.0 to 0.20.0
        - https://github.com/etcd-io/etcd/pull/18403

### CODE REVIEW

**Comment by henrybear327** [comment]
@ivanvc 1.22.6 is out, should I bump the go version in another PR? :)

**Comment by ivanvc** [comment]
> @ivanvc 1.22.6 is out, should I bump the go version in another PR? :)

I created issue #18419 to track this update. Thanks for the heads up, @henrybear327 :)

File: tools/mod/go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module go.etcd.io/etcd/tools/v3
 
-go 1.22.4
```
Comment (ivanvc) [inline tools/mod/go.mod:3]: I found that the dependency that is generating this is: `github.com/alexfalkowski/gocovmerge` (https://github.com/alexfalkowski/gocovmerge/blob/e4976baff208864a31dd6ef35ea755b0cef63030/go.mod#L3). I'll open an issue/pull request in their repository, so we don't have this issue anymore.

**Review by ivanvc — APPROVED** [review]
LGTM. Thanks, @henrybear327

File: tools/mod/go.mod (hunk 1 above)
Comment (ivanvc) [reply tools/mod/go.mod:3]: Opened alexfalkowski/gocovmerge#87.

Comment (henrybear327) [reply tools/mod/go.mod:3]: Merged in lightning speed :D! Thanks @ivanvc

Comment (henrybear327) [reply tools/mod/go.mod:3]: Release done by gocovmerge https://github.com/alexfalkowski/gocovmerge/releases/tag/v1.3.12
Will add a new commit to address this

Comment (henrybear327) [reply tools/mod/go.mod:3]: @ivanvc https://github.com/etcd-io/etcd/pull/18418/commits/39d3e25f73e4d4456c32074c66a7b4243437b2f3

**Comment by ahrtr** [comment]
@henrybear327 Please resolve the workflow failure `pull-etcd-verify`, thx

**Comment by henrybear327** [comment]
> @henrybear327 Please resolve the workflow failure `pull-etcd-verify`, thx

@ahrtr it's resolved! Due to the PR done by @ivanvc and the fast upstream release, I added https://github.com/etcd-io/etcd/commit/39d3e25f73e4d4456c32074c66a7b4243437b2f3 and this caused quite some dependencies to be updated. 

Please see the latest commits that I have posted! Thank you!

**Review by ivanvc — APPROVED** [review]
LGTM on green. Thanks, @henrybear327.
Could you please also update the pull request description to mention the update to `x/net`, `x/crypto`, and `x/text`?

**Comment by henrybear327** [comment]
> LGTM on green. Thanks, @henrybear327. Could you please also update the pull request description to mention the update to `x/net`, `x/crypto`, and `x/text`?

@ivanvc updated as requested. Please take a look! Thank you! :)

**Review by ahrtr — APPROVED** [review]
LGTM

Thanks

**Pull request merged**